crypto/threads_pthread.c

```diff
@@ -274,14 +275,14 @@ static struct rcu_qp *get_hold_current_qp(struct rcu_lock_st *lock)
          * systems like x86, but is relevant on other arches
          * Note: This applies to the reload below as well
          */
-        qp_idx = ATOMIC_LOAD_N(uint32_t, &lock->reader_idx, __ATOMIC_ACQUIRE);
-
+        reader_idx = ATOMIC_LOAD_N(uint32_t, &lock->reader_idx, __ATOMIC_ACQUIRE);
+        qp_idx = reader_idx % lock->group_count;
```

Comment:
when 2^32 modulo group_count != 0, the qp_idx will jump.
I would expect that to cause probems